cpp.mk

```diff
@@ -49,6 +50,12 @@ $${$1_OBJS} : $${$1_OBJECT_DIR}/%.o : %.cpp
 	@mkdir -p $${$1_OBJECT_DIR}
 	${CXX} ${CXXFLAGS} ${CPPFLAGS} -MMD -MP $${$2_CPPFLAGS} $${$1_CPPFLAGS} -c $$< -o $$@
 
+# clang-tidy files are produced by analyzing source files
+$${$1_CLANGTIDY} : $${$1_OBJECT_DIR}/%.clangtidy : %.cpp
+	@mkdir -p $${$1_OBJECT_DIR}
+	clang-tidy-3.8 -checks='*,-cppcoreguidelines-pro-type-vararg,-modernize-use-auto,-cppcoreguidelines-pro-bounds-array-to-pointer-decay' $$< -- ${CXX} ${CXXFLAGS} ${CPPFLAGS} -MMD -MP $${$2_CPPFLAGS} $${$1_CPPFLAGS} > $$@
```

Comment:
It might be nice to have a `$1_COMPILE_COMMAND` variable (that's `=` not `:=` so it's always correct) so we don't have to keep these in sync.